# Анализ автономных некоммерческих организаций получателей федеральных субсидий

Для работы необходимо:
1. Скачать перечень получателей субсидий из федерального бюджета по ссылке https://sub.clearspending.ru/opendata/ файл receivers.zip
2. Распаковать файл receivers.json из архива receivers.zip
3. Загрузить содержимое файла в MongoDB. Команда "mongoimport -d subs -c receivers receivers.json"

Цель этого упражнения: научиться работать с данными субсидий и классифицировать НКО по типу и виду деятельности

Задача: определить число организаций в статусе АНО и не являющихся ВУЗами получавших субсидии и общую сумму полученных ими субсидий.

In [24]:
import json
import pprint
from pymongo import MongoClient
from IPython.display import HTML, display
import tabulate
import pandas as pd

MONGO_SERVER = 'localhost'
MONGO_PORT = 27017


Подключаемся к таблице **receivers** 

In [6]:
client = MongoClient(MONGO_SERVER, MONGO_PORT)
db = client['subs']
receivers = db['receivers']


Делаем выборку организаций имеющих тип АНО

In [50]:
# Поля которые мы хотим получить в итоге при выводе
fields = ['fullName', 'ogrn', 'inn', 'okopfCode', 'okopfName', 'okfsCode', 'okfsName']

# Поля необходимые для запроса в MongoDB (нужен префикс "info." поскольку все они в этом блоке)
query_fields = []
for f in fields:
    query_fields.append('info.' + f)

filtered = {}
for f in query_fields :
    filtered[f] = 1

uniq = []    
table = []
for o in receivers.find({'info.okopfCode' : '71400', }, filtered):
    if o['info']['ogrn'] in uniq: 
        continue
    else:
        uniq.append(o['info']['ogrn'])
    record = []
    for f in fields:
        record.append(o['info'][f])
    table.append(record)

# Смотрим на итоговую таблицу получателей
pd.DataFrame(table, columns=fields)


fullName           ogrn  \
0    АВТОНОМНАЯ НЕКОММЕРЧЕСКАЯ ОРГАНИЗАЦИЯ ДУХОВНО-...  1170327003505   
1    АВТОНОМНАЯ НЕКОММЕРЧЕСКАЯ ОРГАНИЗАЦИЯ ПО СТРОИ...  1175658001518   
2    АВТОНОМНАЯ НЕКОММЕРЧЕСКАЯ ОРГАНИЗАЦИЯ "ИШИМСКИ...  1157232043396   
3    АВТОНОМНАЯ НЕКОММЕРЧЕСКАЯ ОРГАНИЗАЦИЯ "ДУХОВНО...  1177400000238   
4    АВТОНОМНАЯ НЕКОММЕРЧЕСКАЯ ОРГАНИЗАЦИЯ "СТРОИТЕ...  1151832011473   
5    АВТОНОМНАЯ НЕКОММЕРЧЕСКАЯ ОРГАНИЗАЦИЯ "БИЙСКИЙ...  1152225028647   
6    АВТОНОМНАЯ НЕКОММЕРЧЕСКАЯ ОРГАНИЗАЦИЯ "ДУХОВНО...  1175958004848   
7    АВТОНОМНАЯ НЕКОММЕРЧЕСКАЯ ОРГАНИЗАЦИЯ "ДУХОВНО...  1173525001528   
8    АВТОНОМНАЯ НЕКОММЕРЧЕСКАЯ ОРГАНИЗАЦИЯ ПО СТРОИ...  1151001016385   
9    АВТОНОМНАЯ НЕКОММЕРЧЕСКАЯ ОРГАНИЗАЦИЯ "ВОЗРОЖД...  1176000000021   
10   АВТОНОМНАЯ НЕКОММЕРЧЕСКАЯ ОРГАНИЗАЦИЯ "ДУХОВНО...  1151832021120   
11   АВТОНОМНАЯ НЕКОММЕРЧЕСКАЯ ОРГАНИЗАЦИЯ "РУБЦОВС...  1152225025534   
12   АВТОНОМНАЯ НЕКОММЕРЧЕСКАЯ ОРГАНИЗАЦИЯ ПО ОКАЗА...  1144300000965   
13   АВТОНОМНАЯ НЕКОММЕРЧЕСКАЯ ОРГАНИЗАЦИЯ "АГЕНТСТ...  1127799006323   
14   АВТОНОМНАЯ НЕКОММЕРЧЕСКАЯ ОРГАНИЗАЦИЯ ПО ПРЕДО...  1176196047213   
15   АВТОНОМНАЯ НЕКОММЕРЧЕСКАЯ ОРГАНИЗАЦИЯ "ДОБРОВО...  1127799025936   
16   АВТОНОМНАЯ НЕКОММЕРЧЕСКАЯ ОРГАНИЗАЦИЯ "АГЕНТСТ...  1165543054159   
17   АВТОНОМНАЯ НЕКОММЕРЧЕСКАЯ ОРГАНИЗАЦИЯ "МОЛОДЕЖ...  1077799030341   
18   АВТОНОМНАЯ НЕКОММЕРЧЕСКАЯ ОРГАНИЗАЦИЯ "ИНФОРМА...  1147799009380   
19   АВТОНОМНАЯ НЕКОММЕРЧЕСКАЯ ОРГАНИЗАЦИЯ РЕДАКЦИЯ...  1167700059801   
20   АВТОНОМНАЯ НЕКОММЕРЧЕСКАЯ ОРГАНИЗАЦИЯ ВЫСШЕГО ...  1167700069228   
21   АВТОНОМНАЯ НЕКОММЕРЧЕСКАЯ ОРГАНИЗАЦИЯ СТРОИТЕЛ...  1168600050035   
22   АВТОНОМНАЯ НЕКОММЕРЧЕСКАЯ ОБРАЗОВАТЕЛЬНАЯ ОРГА...  1065029009429   
23   АВТОНОМНАЯ НЕКОММЕРЧЕСКАЯ ОРГАНИЗАЦИЯ ВЫСШЕГО ...  1177700003007   
24   АВТОНОМНАЯ НЕКОММЕРЧЕСКАЯ ОРГАНИЗАЦИЯ ВЫСШЕГО ...  1181690004407   
25   АВТОНОМНАЯ НЕКОММЕРЧЕСКАЯ ОРГАНИЗАЦИЯ ВЫСШЕГО ...  1167154074130   
26   АВТОНОМНАЯ НЕКОММЕРЧЕСКАЯ ОРГАНИЗАЦИЯ ВЫСШЕГО ...  1027700482369   
27   АВТОНОМНАЯ НЕКОММЕРЧЕСКАЯ ОБРАЗОВАТЕЛЬНАЯ ОРГА...  1097800000550   
28   АВТОНОМНАЯ НЕКОММЕРЧЕСКАЯ ОРГАНИЗАЦИЯ "РОССИЯ ...  1187700010464   
29   АВТОНОМНАЯ НЕКОММЕРЧЕСКАЯ ОРГАНИЗАЦИЯ "СТУДИЯ ...  1027700355924   
..                                                 ...            ...   
78   АВТОНОМНАЯ НЕКОММЕРЧЕСКАЯ ОРГАНИЗАЦИЯ ВЫСШЕГО ...  1169600000668   
79   АВТОНОМНАЯ НЕКОММЕРЧЕСКАЯ ОРГАНИЗАЦИЯ "АНАЛИТИ...  1157700000655   
80   АВТОНОМНАЯ НЕКОММЕРЧЕСКАЯ ОРГАНИЗАЦИЯ "РЕДАКЦИ...  1024000763214   
81   АВТОНОМНАЯ НЕКОММЕРЧЕСКАЯ ОРГАНИЗАЦИЯ "РЕДАКЦИ...  1084000000248   
82   АВТОНОМНАЯ НЕКОММЕРЧЕСКАЯ ОРГАНИЗАЦИЯ "ТРАНСПО...  1147799018047   
83   АВТОНОМНАЯ НЕКОММЕРЧЕСКАЯ ОРГАНИЗАЦИЯ "РЕДАКЦИ...  1084000001117   
84   АВТОНОМНАЯ НЕКОММЕРЧЕСКАЯ ОРГАНИЗАЦИЯ "РЕДАКЦИ...  1027700131018   
85   ОБРАЗОВАТЕЛЬНАЯ АВТОНОМНАЯ НЕКОММЕРЧЕСКАЯ ОРГА...  1157700006188   
86   АВТОНОМНАЯ НЕКОММЕРЧЕСКАЯ ОРГАНИЗАЦИЯ "ДИРЕКЦИ...  1117799017160   
87   АВТОНОМНАЯ НЕКОММЕРЧЕСКАЯ ОРГАНИЗАЦИЯ "ДИРЕКЦИ...  1137800006156   
88   АВТОНОМНАЯ НЕКОММЕРЧЕСКАЯ ОРГАНИЗАЦИЯ "РЕДАКЦИ...  1145658011311   
89   АВТОНОМНАЯ НЕКОММЕРЧЕСКАЯ ОРГАНИЗАЦИЯ "ЕДИНАЯ ...  1092300000572   
90   ОБРАЗОВАТЕЛЬНАЯ АВТОНОМНАЯ НЕКОММЕРЧЕСКАЯ ОРГА...  1026302005014   
91   ОБРАЗОВАТЕЛЬНАЯ АВТОНОМНАЯ НЕКОММЕРЧЕСКАЯ ОРГА...  1147799014835   
92   АВТОНОМНАЯ НЕКОММЕРЧЕСКАЯ ОРГАНИЗАЦИЯ "ГАЗЕТА ...  1025202614997   
93   АВТОНОМНАЯ НЕКОММЕРЧЕСКАЯ ОРГАНИЗАЦИЯ "СТУДИО-...  1047796462889   
94   АВТОНОМНАЯ НЕКОММЕРЧЕСКАЯ ОРГАНИЗАЦИЯ "РЕДАКЦИ...  1047709009974   
95   АВТОНОМНАЯ НЕКОММЕРЧЕСКАЯ ОРГАНИЗАЦИЯ ПЕРИОДИЧ...  1087799010529   
96   АВТОНОМНАЯ НЕКОММЕРЧЕСКАЯ ОРГАНИЗАЦИЯ "РЕДАКЦИ...  1027739357084   
97   АВТОНОМНАЯ НЕКОММЕРЧЕСКАЯ ОРГАНИЗАЦИЯ "ТВ-НОВО...  1057746595367   
98   АВТОНОМНАЯ НЕКОММЕРЧЕСКАЯ ОРГАНИЗАЦИЯ "ИСПОЛНИ...  1091600002922   
99   АВТОНОМНАЯ НЕКОММЕРЧЕСКАЯ ОРГАНИЗАЦИЯ "ЦЕНТР С...  1077799005877   
100  АВТОНОМНАЯ НЕКОММЕРЧЕСКАЯ ОРГАНИЗАЦИЯ ВЫСШЕГО ...  1157700015846   
101  АВТОНОМНАЯ НЕКОММЕРЧЕСКАЯ 

In [ ]:
# Собираем сведения об объёме полученных субсидий и складываем их в новую коллекцию
import requests as rq

db.drop_collection('subsidies')
subsidies = db['subsidies']

n = 0
for o in table:
    n += 1
    print('%d. Собираем данные по %s' % (n, o[0]) )
    summ = 0
    page = 1
    while True:
        url = "https://api.sub.clearspending.ru/v1/subsidy/?receiver_inn=" + o[2]+'&page=%d' % (page)
        rec = rq.get(url).json()
        for s in rec['items']:
            subsidies.save(s)
        if rec['count'] == 50:            
            page += 1            
        else:
            break




На основе собранной коллекции подсчитываем общую сумма бюджета на субсидии именно АНО в миллиардах рублей

In [63]:
total_summ = 0
for o in subsidies.find():
    total_summ += o['info']['currencySum']

final = float(total_summ) / 1000000000
print('Всего перечислено федеральных субсидий АНО: %d млрд. руб.' % (final))
    

Всего перечислено федеральных субсидий АНО: 155 млрд. руб.


А теперь попробуем найти топ получателей и сколько получил каждый их них

In [96]:
m_table = []
for o in table:
    summ = 0
    for s in subsidies.find({'receiver.inn' : o[2]}):
        summ += s['info']['currencySum']    
    m_table.append([o[0], o[2], summ])
    
df = pd.DataFrame(m_table, columns=['name', 'inn', 'summ'])
sdf = df.sort_values(by=['summ'], ascending=False)
out_table = [['млрд. рублей', "организация"]]
for o in sdf.values:
    out_table.append([o[2] / 1000000000, o[0]])

display(HTML(tabulate.tabulate(out_table, tablefmt='html')))
#print(df.nlargest(2, 'summ').sum().values[2] / 1000000000)

Попробуем разобраться как это соотносится с деятельностью организаций. Получим основные коды ОКВЭД каждой организации получившей более 100 миллионов рублей


In [102]:
out_table = [['млрд. рублей', "код деятельности", "деятельность", "организация"]]
for o in sdf[(sdf['summ']>100000000)].values:
    i = receivers.find_one({'info.inn' : o[1]})
    out_table.append([o[2] / 1000000000, i['activities'][0]['activityCode'], i['activities'][0]['activityName'], o[0]])
    
display(HTML(tabulate.tabulate(out_table, tablefmt='html')))


Из перечисленных некоммерческих организаций мы видим 3 университета: Российский новый университет, Сколтех и Гуманитарный университет которые получают субсидии как организации высшего образования. Остальные организации это специально созданные организации по выполнению тех или иных государственных задач. Отфильтруем только их и посмотрим сколько средств уходит на подобные организации в форме АНО. Добавим в текущий DataFrame данные по кодам деятельности организаций и отфильтруем те организации которые не относятся к высшему образованию.

In [107]:
out_table = [['млрд. рублей', "код деятельности", "деятельность", "организация"]]
act_col_code = []
act_col_name = []
for o in sdf.values:
    i = receivers.find_one({'info.inn' : o[1]})
    act_col_code.append( i['activities'][0]['activityCode'])
    act_col_name.append( i['activities'][0]['activityName'])
sdf['activityCode'] = act_col_code
sdf['activityName'] = act_col_name
    
display(HTML(tabulate.tabulate(sdf[(sdf['summ']>100000000) & (sdf['activityCode'] != '82.22')], tablefmt='html')))

    В завершение подсчитаем сколько всего организаций в статусе АНО и и не являющихся ВУЗами получили субсидий из федерального бюджета и на какую сумму

In [117]:
filtered_df = sdf[(sdf['summ']>100000000) & (sdf['activityCode'] != '82.22')]
print('Всего получателей субсидий в форме АНО (кроме ВУЗов) и получивших от 100 млн. рублей: %d' % (len(filtered_df.values)))


print('Общая сумма: %f млрд рублей' % (filtered_df.sum().values[2] / 1000000000))


Всего получателей субсидий в форме АНО (кроме ВУЗов) и получивших от 100 млн. рублей: 20
Общая сумма: 154.348284 млрд рублей


Итого, на конец марта 2019 года, 20 организаций в форме АНО и не являющихся ВУЗами получили 154 миллиарда рублей. 
Тем самым 154 миллиарда рублей были выведены из под законов 44-ФЗ и 223-ФЗ о госзакупках.

Вопросы и предложения направляйте на ibegtin@infoculture.ru